dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/storage/GitHubJsonStore.java

```diff
@@ -0,0 +1,91 @@
+package com.salesforce.dockerfileimageupdate.storage;
+
+import com.google.gson.*;
+import com.salesforce.dockerfileimageupdate.utils.Constants;
+import com.salesforce.dockerfileimageupdate.utils.GitHubUtil;
+import org.kohsuke.github.GHContent;
+import org.kohsuke.github.GHMyself;
+import org.kohsuke.github.GHRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.file.Paths;
+
+public class GitHubJsonStore {
+    private static final Logger log = LoggerFactory.getLogger(GitHubJsonStore.class);
+    private final GitHubUtil gitHubUtil;
+    private final String store;
+
+    public GitHubJsonStore(GitHubUtil gitHubUtil, String store) {
+        this.gitHubUtil = gitHubUtil;
+        this.store = store;
+    }
+
+    /* The store link should be a repository name on Github. */
+    public void updateStore(String img, String tag) throws IOException {
+        if (store == null) {
+            log.info("Image tag store cannot be null. Skipping store update...");
+            return;
+        }
+        log.info("Updating store: {} with image: {} tag: {}...", store, img, tag);
+        GHRepository storeRepo;
+        try {
+            GHMyself myself = gitHubUtil.getMyself();
+            String ownerOrg = myself.getLogin();
+            storeRepo = gitHubUtil.getRepo(Paths.get(ownerOrg, store).toString());
+        } catch (IOException e) {
+            storeRepo = gitHubUtil.createPublicRepo(store);
+        }
+        updateStoreOnGithub(storeRepo, Constants.STORE_JSON_FILE, img, tag);
+    }
+
+    protected void updateStoreOnGithub(GHRepository repo, String path, String img, String tag) throws IOException {
+        try {
+            repo.getFileContent(path);
+        } catch (IOException e) {
+            repo.createContent().content("").message("initializing store").path(path).commit();
+        }
+
+        String latestCommit = repo.getBranches().get(repo.getDefaultBranch()).getSHA1();
+        log.info("Loading image store at commit {}", latestCommit);
+        GHContent content = repo.getFileContent(path, latestCommit);
+
+        if (content.isFile()) {
+            JsonElement json;
+            try (InputStream stream = content.read(); InputStreamReader streamR = new InputStreamReader(stream)) {
+                try {
+                    json = JsonParser.parseReader(streamR);
+                } catch (JsonParseException e) {
+                    log.warn("Not a JSON format store. Clearing and rewriting as JSON...");
+                    json = JsonNull.INSTANCE;
+                }
+            }
+            String jsonOutput = getAndModifyJsonString(json, img, tag);
+            content.update(jsonOutput,
```

Comment:
Doe this include retries?

Reply:
Nah, it doesn't currently. I haven't observed many issues where retries set this in a bad spot. That is something that could be considered for sure. I would have a similar concern about determining when to retry to help reduce rate limit pressure.